### PR TITLE
feat: RLS 

### PR DESCRIPTION
**Pull request by adityathebe**

### CODE REVIEW

File: models/permission_test.go [hunk 1]
```diff
@@ -33,6 +34,13 @@ func TestPermission_Condition(t *testing.T) {
 			perm:     Permission{},
 			expected: "",
 		},
+		{
+			name: "agents",
```
Comment (moshloop) [inline models/permission_test.go:38]: Don't we need one for tags

File: tests/rls_test.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+				}),
+				Entry("correct agent", testCase{
+					name:          "correct agent",
+					jwtClaims:     `{"tags": {"cluster": "testing-cluster"}, "agents": ["00000000-0000-0000-0000-000000000000"]}`,
+					expectedCount: &totalConfigs,
+				}),
+				Entry("correct tag", testCase{
+					name:          "correct tag",
+					jwtClaims:     `{"tags": {"account": "flanksource"}, "agents": ["10000000-0000-0000-0000-000000000000"]}`,
+					expectedCount: &numConfigsWithFlanksourceTag,
```
Comment (moshloop) [inline tests/rls_test.go:73]: Are these not going to be 0 as no test data is inserted ?

Comment (adityathebe) [reply tests/rls_test.go:73]: the testdata are the dummy configs we have inn suite setup

**Comment by moshloop** [comment]
@adityathebe Can we benchmark this @ 10k, , 25k, 50k, and 100k and add the results with and without RLS to a BENCHMARK.md ?

**Pull request merged**